workflows/ci_test_dashboard/tests/test_generate_ui.py

```
@@ -0,0 +1,175 @@
+
+        # Ensure that each GPU version appears only once in the HTML.
+        self.assertEqual(html.count("24.6.2"), 1)
+        self.assertEqual(html.count("24.9.2"), 1)
+        # Verify that the deduplication logic selected the entry with the latest job_timestamp.
+        self.assertIn('href="https://example.com/1-dup"', html)
+        # Ensure that the entry with test_status FAILURE (gpu "25.0.0") does not appear.
+        self.assertNotIn("25.0.0", html)
+
+
+if __name__ == '__main__':
+    unittest.main()
```

Ruff (0.8.2)

Undefined name `unittest`

(F821)